### PR TITLE
Maintenance: Upgrade Provenance and Figure Libs + QoL

### PR DESCRIPTION
**Pull request by hyperschwartz**
# Description
This PR upgrades to the latest Provenance Protos + PbClient, as well as moving to Figure Tech event stream + Figure Tech OS Gateway libs.  Additionally, it adds a precommit hook to ensure lints have run.

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,89 +1,90 @@
 [versions]
-assetModel = "0.1.2"
-assetSpec = "1.0.0"
-bouncyCastle = "1.70"
+asset-model = "0.1.2"
```
Comment (hyperschwartz) [inline gradle/libs.versions.toml:2]: A few "real" changes here, but mostly this renames the dependencies to a more better format.  The effects can be seen in all the affected `build.gradle.kts` files across the application.  It's mostly an aesthetic thing.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,89 +1,90 @@
...
+asset-specs = "1.0.0"
+blockapi = "0.1.3"
+bouncycastle = "1.70"
 feign = "11.8"
+figure-eventstream = "0.7.1"
```
Comment (hyperschwartz) [inline gradle/libs.versions.toml:7]: Real change: Bumped to the new eventstream libs from figure.tech versus provenance.io

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -1,89 +1,90 @@
...
 protobuf = "3.21.1"
-provenanceClient = "1.1.1"
-provenanceEventStream = "0.6.3"
-provenanceGateway = "2.0.2"
-provenanceHdWallet = "0.1.15"
-provenanceProto = "1.10.0"
-provenanceScope = "0.6.0"
+provenance-client = "1.3.0"
+provenance-hdwallet = "0.1.15"
+provenance-proto = "1.13.0"
```
Comment (hyperschwartz) [inline gradle/libs.versions.toml:21]: Real change: Upgraded to v1.13.0 for provenance protos

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -1,89 +1,90 @@
...
+objectstore-gateway = "3.1.2"
+okhttp = "4.9.3"
 protobuf = "3.21.1"
-provenanceClient = "1.1.1"
-provenanceEventStream = "0.6.3"
-provenanceGateway = "2.0.2"
-provenanceHdWallet = "0.1.15"
-provenanceProto = "1.10.0"
-provenanceScope = "0.6.0"
+provenance-client = "1.3.0"
```
Comment (hyperschwartz) [inline gradle/libs.versions.toml:19]: Real change: Upgraded to v1.3.0 for PbClient to access the groups goodness in downstream consumers without having to re-import pbClient

**Review by piercetrey-figure — APPROVED** [review]

**Pull request merged**